rules/firestore.rules

```diff
@@ -50,14 +50,8 @@ service cloud.firestore {
       return request.auth.token.email in get(/databases/$(database)/documents/projects/$(anyProject)/invitation_lists/main).data.invites;
     }
 
-    // Since firebase doesn't have the ability to specify what CAN be updated
-    // We need to say what CANNOT be updated - that is all of the tasks fields BUT listeners
     function isNotUpdatingAnythingButListeners() {
-      return notUpdating('id') && notUpdating('title') && notUpdating('assignee')
-      && notUpdating('type') && notUpdating('projectName') && notUpdating('label')
-      && notUpdating('created') && notUpdating('dueDate') && notUpdating('creator')
-      && notUpdating('isDone') && notUpdating('doneDate') && notUpdating('description')
-      && notUpdating('requirements') && notUpdating('isCritical') && notUpdating('extraFields');
+      return request.resource.data.size() == 1 && request.resource.data.keys().hasOnly(['listeners'])
```

Comment:
@rootux let me know if you agree here. the tests are good with that

Reply:
No. This is not good. Please read the stack overflow link I've commented just above that line.t. This was the old solution we used to have. It's not working

Reply:
Read the comment under notUpdating function

Reply:
1. I understand and will revert.
2. I'm curious, if it is the case and request.size should be as the next document state (having all the existing fields) how come on  the test of the listeners, the test passes when I send only listeners, check that the size of the request is 1, but the existing doc is bigger then one field (u can see my console log on line 384 if u fetch the branch)

Reply:
the tests pass also on by your implementation.
@rootux I want to be able to write a test that will fail the wrong implementation

Reply:
As for 2. That might be due to different behavior on emulator and production.
I've talked with the firebase team and they are working on a proper solution

Reply:
wow amazing! can u frw me the email with their response